contrib/enfuseAdvanced.lua

```diff
@@ -612,434 +210,851 @@ local function GetFileName(full_path)
+			GUI.Target.on_conflict.selected = 1
+			GUI.Target.on_conflict.sensitive = false
+			GUI.Presets.variants_type.sensitive = true
+		else
+			GUI.Target.on_conflict.sensitive = true
+			GUI.Presets.variants_type.sensitive = false
+		end
+	end,
+	reset_callback = function(self) self.value = false end
+}
+temp = dt.preferences.read(mod, 'active_variants_type_ind', 'integer')
+if not InRange(temp, 1, 2) then temp = 1 end
+GUI.Presets.variants_type = dt.new_widget('combobox'){
+	label = "create variants type",
+	tooltip = "preset type to be used when creating image variants",
+	selected = temp,
+	'dri', 'dff',
+	changed_callback = function(self)
+		dt.preferences.write(mod, 'active_variants_type', 'string', self.value) 
+		dt.preferences.write(mod, 'active_variants_type_ind', 'integer', self.selected)
+	end,
+	reset_callback = function(self) 
+		self.selected = 1
+		dt.preferences.write(mod, 'active_variants_type', 'string', self.value) 
+		dt.preferences.write(mod, 'active_variants_type_ind', 'integer', self.selected)
+	end	
+}
+GUI.Presets.variants_type.sensitive = GUI.Presets.variants.value
+temp = GUI.Target.format.value
+if temp == 'tif' then temp = 1
+elseif temp == 'jpg' then temp = 2
+else temp = 3
+end
+stack_compression.active = temp
+for _,widget in pairs(GUI.AIS) do
+	widget.sensitive = GUI.align.value
 end
-local enfuse_path_widget = dt.new_widget("file_chooser_button"){
-	title = "Select enfuse[.exe] file",
-	value = enfuse_path,
-	is_directory = false,
+GUI.Target.apply_style.sensitive = GUI.Target.auto_import.value
+GUI.Target.copy_tags.sensitive = GUI.Target.auto_import.value
+GUI.Target.add_tags.sensitive = GUI.Target.auto_import.value
+
+local box_AIS = dt.new_widget('box'){
+	orientation = 'vertical',
+    label_AIS_options,
+	GUI.align,
+	GUI.AIS.radial_distortion,
+	GUI.AIS.optimize_field,
+	GUI.AIS.optimize_image_center,
+	GUI.AIS.auto_crop,
+	GUI.AIS.distortion,
+	GUI.AIS.gpu,
+	GUI.AIS.grid_size,
+	GUI.AIS.control_points,
+	GUI.AIS.control_points_remove,
+	GUI.AIS.correlation
+}
+local box_ENF = dt.new_widget('box'){
+	orientation = 'vertical',
+	label_ENF_options,
+	GUI.ENF.exposure_weight,
+	GUI.ENF.saturation_weight,
+	GUI.ENF.contrast_weight,
+	GUI.ENF.exposure_optimum,
+	GUI.ENF.exposure_width,
+	GUI.ENF.hard_masks,
+	GUI.ENF.save_masks,
+	GUI.ENF.contrast_window_size,
+	GUI.ENF.contrast_edge_scale,
+	GUI.ENF.contrast_min_curvature,
+	GUI.Presets.current_preset,
+	GUI.Presets.load,
+	GUI.Presets.save
+}
+local box_Target = dt.new_widget('box'){
+	orientation = 'vertical',
+	label_target_options,
+	GUI.Target.format,
+	GUI.Target.depth,
+	stack_compression,
+	label_directory,
+	GUI.Target.output_directory,
+	GUI.Target.source_location,
+	GUI.Target.on_conflict,
+	GUI.Presets.variants,
+	GUI.Presets.variants_type,
+	GUI.Target.auto_import,
+	GUI.Target.apply_style,
+	GUI.Target.copy_tags,
+	GUI.Target.add_tags
+}
+GUI.options_contain = dt.new_widget('stack'){
+	active = 2,
+	box_AIS,
+	box_ENF,
+	box_Target
+}
```

Comment:
You could add a configuration stack for the executables, and if it's the first run then make that the default.

Reply:
That's a great idea. Then if it detects an install error I can make that container active in the stack.